app/frontend/controllers/conditional_filter_controller.js

```diff
@@ -0,0 +1,13 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  initialize() {
+    this.clearIfHidden()
+  }
+
+  clearIfHidden() {
+    if(this.element.style["display"] == "none") {
+      this.element.value = ""
+    }
+  }
```

Comment:
should this have a unit test for the js?

Reply:
Yep, that's what this test is for https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/582/files#diff-d319a8fedb48067418a804668bdd086508689cb5ec51d54525788f418c32c705R529